### PR TITLE
SITL: Adding an error message output condition

### PR DESCRIPTION
**Pull request by muramura**
If it fails to send or receive data, it gets -1.
I would suggest printing out an error message and aborting the process if it fails to send or receive.

### CODE REVIEW

File: libraries/SITL/SIM_I2C.cpp [hunk 1]
```diff
@@ -93,7 +93,7 @@ int I2C::ioctl_rdwr(i2c_rdwr_ioctl_data *data)
                 handled = true;
             }
         }
-        if (!handled) {
+        if (!handled || ret) {
```
Comment (khancyr) [inline libraries/SITL/SIM_I2C.cpp:96]: it should be !ret or ret < 0 if you want to catch the error

**Review by khancyr — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
I'm not sure we do want to fiddle with ret in this manner; doing so would preclude synthesising bad io read/writes returned by the operating system (e.g. on Linux).  Canonicalising it to -1 doesn't make sense to me ATM.

**Comment by muramura** [comment]
@peterbarker san. 
This is the process of sending to multiple devices.
For this reason, the result of the last device is returned.
This is not correct.

File: libraries/SITL/SIM_I2C.cpp (hunk 1 above)
Comment (muramura) [reply libraries/SITL/SIM_I2C.cpp:96]: @khancyr san.
This method dev_at_address.device.rdwr(data) has a normal value of 0.
Therefore, non-zero is an abnormal value.

Comment (peterbarker) [reply libraries/SITL/SIM_I2C.cpp:96]: @muramura Oh, I see.  In that case the sanity check should be on line 92 - if handled is already set then raise hell.

**Comment by peterbarker** [comment]
@muramura I've solved this in a different fashion by adding a sanity check that we only have a single simulated device at a bus/addr combo.

That PR is here: https://github.com/ArduPilot/ardupilot/pull/16209

Thanks for finding this one - it was a silly bug.

**Pull request closed without merging**